### DISCUSSION TITLE
How do I generate the xxxConfiguration class

### DISCUSSION THREAD

**Discussion by z-z-liu** [smallrye/smallrye-stork#263] [General]
I according to the document "http://smallrye.io/smallrye-stork/1.1.0/custom-service-discovery/" custom "Service Discovery", there is no automatic generating xxxConfiguration class, Additional configuration required?

My parent pom is not “stork-parent”，is that the reason？

**Comment by aureamunoz** [comment]
Hi @liuzhaozhao !

To set up the configuration for your custom service discovery , you need to annotate your CustomServiceDiscoveryProvider with `@ServiceDiscoveryType` and `@ServiceDiscoveryAttribute`. This will generate the needed classes `CustomServiceDiscoveryProviderConfiguration`. For e.g. look at the [Stork Consul Service Discovery provider](https://github.com/smallrye/smallrye-stork/blob/326bbe71077fd537ed879d487d7004e7d5abf7e1/service-discovery/consul/src/main/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryProvider.java#L11)
I will transfer this one to a discussion because it's not an issue.

I hope that helps

**Reply by z-z-liu** [reply]
I did it the way you said, but there is still no automatic generation of configuration classes
![image](https://user-images.githubusercontent.com/10733976/158564042-7bb33ddd-b563-4c11-bb85-89ff96f34740.png)
![image](https://user-images.githubusercontent.com/10733976/158564979-f5437d72-3ea1-458a-8be7-c61652fc9f60.png)
![image](https://user-images.githubusercontent.com/10733976/158565081-64fd9272-4948-4c7c-bb5b-e5b9362a0cd7.png)


I used the nacos-based service discovery plug-in developed by 1.0.0.Beta1, but it changed a lot after upgrade 1.1.0. The main problem is that configuration classes cannot be generated automatically.

I use IDEA as the development tool. Is there anything else missing?

**Comment by aureamunoz** [comment]
Hi, the annotations and the `stork-configuration-generator` should do the job. What version of Stork are you using? do you have an error in the console?

**Reply by z-z-liu** [reply]
The version of Stork used in the screenshot above is 1.1.0
The error message is that the class cannot be found.
![image](https://user-images.githubusercontent.com/10733976/158617404-592223c0-45cb-42c3-b57d-7714422f184a.png)

Is this class automatically generated or does maven need to be executed?

I according to the **stork-service-discovery-consul** way of writing, still have a question, and, in the form of maven in idea to import the https://github.com/smallrye/smallrye-stork.git project, Run time is also can't find ConsulServiceDiscoveryProviderConfiguration
![image](https://user-images.githubusercontent.com/10733976/158617913-727a9962-1af1-4143-bafe-78a6531d1368.png)



I'm not sure how to use Stork-configuration-generator to generate this class

**Comment by aureamunoz** [comment]
Ok, so the Stork version is ok. You need to perform maven build in order to generate the class. Try with `mvn clean install` from the root of the project. The class should be generated under the `target/generated-sources/annotations/` folder.

Sometimes in IDEA you need to recharge/reimport Maven Projects also

**Reply by z-z-liu** [reply]
**stork-service-discovery-consul**  generates a Configuration file, but generates it in the **target\generated-sources\annotations** directory instead of **ConsulServiceDiscoveryProvider** source directory, causing the class file not to be found
![image](https://user-images.githubusercontent.com/10733976/158633673-ac2746d1-bf46-4af1-87fe-d09dd7572791.png)

**Reply by aureamunoz** [reply]
> **stork-service-discovery-consul** generates a Configuration file, but generates it in the **target\generated-sources\annotations** directory instead of **ConsulServiceDiscoveryProvider** source directory, causing the class file not to be found ![image](https://user-images.githubusercontent.com/10733976/158633673-ac2746d1-bf46-4af1-87fe-d09dd7572791.png)

That's correct. This is how it works. What about your custom service discovery? Have you been able to generate it?

**Reply by z-z-liu** [reply]
My custom service discovery that the configuration was not generated and I created the file manually.

I wrote the Quarkus plug-in, which may be related to this, because I did not investigate specific problems due to the time problem.

**Comment by k33g** [comment]
I have the same problem: "there is no automatic generating xxxConfiguration class". I may have missed an obvious step, but step-by-step guidance would be excellent. (I'm using the `1.4.1` version of Stork)

**Reply by k33g** [reply]
Update:
- Source Generated
- Maven project reimported

You should add some guidance to the documentation or a link to this discussion.

**Reply by aureamunoz** [reply]
Hi!
Thanks for the feedback. I created an [issue](https://github.com/smallrye/smallrye-stork/issues/463) to add something useful in the docs.
